runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -711,20 +703,14 @@ public void invalidateClientCache() {
     }
 
     /**
-     * Check list of returned data for trimmed data.
-     *
-     * @param result list of read data
-     *
-     * @return map with valid data (not trimmed) or TrimmedException is thrown is data is trimmed and
-     * flag is not set.
+     * Cache list of returned data with given cache options.
      */
-    private Map<Long, ILogData> checkForTrimmedAddresses(Map<Long, ILogData> result, ReadOptions options) {
-        List<Long> trimmedAddresses = new ArrayList<>();
-
+    private Map<Long, ILogData> checkForCompactedAddresses(Map<Long, ILogData> result, ReadOptions options) {
+        Collection<Long> compactedAddressses = new ArrayList<>();
         for (Map.Entry<Long, ILogData> entry : result.entrySet()) {
-            // Add trimmed addresses to list
-            if (!checkLogData(entry.getKey(), entry.getValue(), false)) {
-                trimmedAddresses.add(entry.getKey());
+            // Add compacted addresses to list
+            if (!checkLogData(entry.getKey(), entry.getValue())) {
+                compactedAddressses.add(entry.getKey());
```

Comment:
Once we detect compacted addresses, shouldn't we remove them from the resolvedQueue? that way we clean up (do the gc functionality that we used to do on these queues before)

Reply:
Yes. Once one address is detected as compacted, it should be not saved in the stream layer as well as address space cache.